### PR TITLE
mq_enabled_chk: Add multi-queue check for virtio-net on windows guest

### PR DESCRIPTION
**Pull request by kimi1978wy**
Add windows support for mq_check script

Signed-off-by: Yu Wang <wyu@redhat.com>

id: 1566965

### CODE REVIEW

**Comment by kimi1978wy** [comment]
(1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.mq_enabled_chk: PASS (122.41 s)
 (1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.mq_enabled_chk: PASS (320.93 s)

File: qemu/tests/cfg/mq_enabled_chk.cfg [hunk 1]
```diff
@@ -8,6 +7,13 @@
     type = mq_enabled_chk
     vhost = on
     image_snapshot = yes
-    smp = 4
-    queues_list = 8 4 2
-    start_vm = no
+    timeout = 360
+    driver_name = "netkvm"
```
Comment (ybduan) [inline qemu/tests/cfg/mq_enabled_chk.cfg]: This can be moved to windows part.

File: qemu/tests/cfg/mq_enabled_chk.cfg [hunk 2]
```diff
@@ -8,6 +7,13 @@
...
+    Linux:
+        start_vm = no
+        smp = 4
+        queues_list = 8 4 2
+    Windows:
+        type = mq_enabled_chk_windows
+        queues = ${smp}
```
Comment (ybduan) [inline qemu/tests/cfg/mq_enabled_chk.cfg]: Does this smp refer to the smp defined in Linux part?

**Comment by kimi1978wy** [comment]
(1/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.boot_with_max_queues: PASS (226.28 s)
 (2/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.boot_with_max_queues: PASS (542.21 s)

File: qemu/tests/cfg/mq_enabled_chk.cfg (hunk 1 above)
Comment (kimi1978wy) [reply qemu/tests/cfg/mq_enabled_chk.cfg]: will modified, thanks for reminder

File: qemu/tests/cfg/mq_enabled_chk.cfg (hunk 2 above)
Comment (kimi1978wy) [reply qemu/tests/cfg/mq_enabled_chk.cfg]: For linux part, this case need different queues to assign, but for windows we need not change queues (smp=queues), so need not to fix the smp value.

File: qemu/tests/mq_enabled_chk_windows.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    msis, queues = utils_net.get_MSIs_and_queues_windows(params, vm)
+    if not msis or not queues:
+        test.fail("Can't get msi & queues info")
+    nic = vm.virtnet[0]
+    nic_queues = int(nic['queues'])
+    nic_vectors = int(nic['vectors']) if nic['vectors'] else (2*queues + 2)
+    if nic_vectors != 2 * nic_queues + 2:
+        # custom vectors
+        nic_queues = (nic_vectors - 2) / 2
```
Comment (hereischen) [inline qemu/tests/mq_enabled_chk_windows.py]: please use double `//`

File: qemu/tests/mq_enabled_chk_windows.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                                                            test,
+                                                            driver_name,
+                                                            timeout)
+
+    msis, queues = utils_net.get_MSIs_and_queues_windows(params, vm)
+    if not msis or not queues:
+        test.fail("Can't get msi & queues info")
+    nic = vm.virtnet[0]
+    nic_queues = int(nic['queues'])
+    nic_vectors = int(nic['vectors']) if nic['vectors'] else (2*queues + 2)
```
Comment (hereischen) [inline qemu/tests/mq_enabled_chk_windows.py]: space between `2 * queues`

File: qemu/tests/mq_enabled_chk_windows.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    if not msis or not queues:
+        test.fail("Can't get msi & queues info")
+    nic = vm.virtnet[0]
+    nic_queues = int(nic['queues'])
+    nic_vectors = int(nic['vectors']) if nic['vectors'] else (2 * queues + 2)
+    if nic_vectors != 2 * nic_queues + 2:
+        # custom vectors
+        nic_queues = (nic_vectors - 2) // 2
+        if nic_queues == 0:
+            nic_queues = 1
```
Comment (vivianQizhu) [inline qemu/tests/mq_enabled_chk_windows.py]: The check looks very similar with this one: https://github.com/autotest/tp-qemu/commit/4200af4be92ed23fd7f17394f0d83476791fac4e#diff-fbb86110c111aabbe176bb0ea72ef488R84 Just the configuration of queues are different, so I suppose this test can reuse the previous script just with different cfg configuration.

**Comment by kimi1978wy** [comment]
(1/2) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.boot_with_max_queues: PASS (542.80 s)
 (2/2) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.mq_enabled_chk: PASS (579.64 s)

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

File: qemu/tests/boot_with_multiqueue.py [hunk 6]
```diff
@@ -23,7 +23,7 @@ def run(test, params, env):
     login_timeout = int(params.get("login_timeout", 360))
     cmd_timeout = int(params.get("cmd_timeout", 240))
 
-    # boot the vm with the max queues
+    # boot the vm with the  queues
```
Comment (vivianQizhu) [inline qemu/tests/boot_with_multiqueue.py]: Extra blank before `queues`

**Pull request merged**